src/imap/mod.rs

```diff
@@ -430,19 +430,17 @@ impl Imap {
         });
     }
 
-    pub fn fetch(&self, context: &Context, watch_folder: &str) -> Result<()> {
-        task::block_on(async move {
-            if !context.sql.is_open() {
-                // probably shutdown
-                return Err(Error::InTeardown);
-            }
-            self.setup_handle_if_needed(context).await?;
+    pub async fn fetch(&self, context: &Context, watch_folder: &str) -> Result<()> {
```

Comment:
hum, so far we don't expose public async api i think but maybe it's ok to start doing that.